### PR TITLE
Byte inversion fix

### PR DESCRIPTION
**Pull request by camilbancioiu**

### CODE REVIEW

File: arwen/elrondapi/elrondei.go [hunk 1]
```diff
@@ -891,8 +888,7 @@ func executeOnSameContext(
 	gasToUse += metering.GasSchedule().BaseOperationCost.DataCopyPerByte * uint64(actualLen)
 	metering.UseGas(gasToUse)
 
-	invBytes := arwen.InverseBytes(value)
-	bigIntVal := big.NewInt(0).SetBytes(invBytes)
+	bigIntVal := big.NewInt(0).SetBytes(value)
```
Comment (sasurobert) [inline arwen/elrondapi/elrondei.go]: I am not sure that these api's would work without inversion.

Comment (camilbancioiu) [reply arwen/elrondapi/elrondei.go]: Added tests specifically for `getCallValue()` and `int64getArguments()`. Tests for `executeOn(Same|Dest)Context()` are on the execution tests branch.

File: arwen/host/elrondei_test.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+	// 1-byte call value
+	host, _ := DefaultTestArwenForCall(t, code)
+	input := DefaultTestContractCallInput()
+	input.GasProvided = 100000
+	input.Function = "test_getCallValue_1byte"
+	input.CallValue = big.NewInt(64)
+
+	vmOutput, err := host.RunSmartContractCall(input)
+	assert.Nil(t, err)
```
Comment (andreibancioiu) [inline arwen/host/elrondei_test.go]: Use `require` instead of `assert`.

File: test/contracts/elrondei/elrondei.c [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+	} else {
+		finish(msg_not_ok, 6);
+	}
+
+	finish((byte*)&length, 4);
+	finish(value, length);
+}
+
+void test_getCallValue_bigInt_to_Bytes() {
+	int length = getCallValue(value);
```
Comment (andreibancioiu) [inline test/contracts/elrondei/elrondei.c]: Add extra test to compare (assert equality) the value from `bigIntGetCallValue` and a big int obtained initialized with the bytes from `getCallValue`.

File: test/contracts/elrondei/elrondei.c [hunk 4]
```diff
@@ -0,0 +1,88 @@
+#include "../elrond/context.h"
+
```
Comment (andreibancioiu) [inline test/contracts/elrondei/elrondei.c]: Add extra test to call `executeOnSameContext` to another function in another contract, function which takes the call value and calls `transferValue` / 2 to some hard-coded address.

Comment (camilbancioiu) [reply test/contracts/elrondei/elrondei.c]: Will come as a next PR, after `executeOnSameContext` is fixed.

File: arwen/helpers.go [hunk 5]
```diff
@@ -50,6 +50,23 @@ func GuardedGetBytesSlice(data []byte, offset int32, length int32) ([]byte, erro
...
+	if len(data) == 0 {
+		return []byte{}
+	}
+	padSize := size - len(data)
+	if padSize <= 0 {
+		return data
+	}
+
+	paddedBytes := make([]byte, padSize)
+	paddedBytes = append(paddedBytes, data...)
```
Comment (sasurobert) [inline arwen/helpers.go]: why is this used ?

**Pull request merged**